Sources/Datadog/Core/System/Time/ServerDateCorrector.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Calculates the date correction for adjusting device time to server time.
+internal protocol DateCorrector {
+    /// Returns recent date correction for adjusting device time to server time.
+    var offset: TimeInterval { get }
+}
+
+internal class ServerDateCorrector: DateCorrector {
+    /// Server offset publisher.
+    private let publisher: ValuePublisher<TimeInterval?> = ValuePublisher(initialValue: nil)
+
+    init(serverDateProvider: ServerDateProvider) {
```

Comment:
`deviceDateProvider` was only used for the console to print the current time. It's best to remove that dependency.

Reply:
Sounds good 👍. So the V2 Logging feature will use its own device time provider for console logs and leverage NTP time for remote logging, right?

Reply:
No, I mean the previous `DateCorrector` was expecting a `DeviceDateProvider` at init, see [here](https://github.com/DataDog/dd-sdk-ios/pull/913/files#diff-69ca19ff72791841d0de0246d6cc7ed0551e2ca932d142cc4a0ca84374cdd82bL36-L38)

But this parameter was only used to print:
```
DD.logger.error(
    """
    NTP time synchronization failed.
    Device time will be used for signing events (current device time is \(deviceTime)).
    """
)
```

Reply:
Ah, got it 👍. Looks good, printing device time is not a big deal 👌 (it will appear in `DD.logger.error()` anyway).